image_proc/test/test_rectify.cpp

```diff
@@ -122,37 +129,49 @@ class ImageProcRectifyTest
     }
 
     raw_image_ = cv_bridge::CvImage(
-      std_msgs::Header(), "bgr8", distorted_image_).toImageMsg();
+      std_msgs::msg::Header(), "bgr8", distorted_image_).toImageMsg();
+
+    rclcpp::NodeOptions options;
+    options.arguments({
+      "--ros-args", "-r", std::string("__ns:=") + "/camera",
+      "-r", "/camera/image:=/camera/image_raw"});
+    node_rectify = std::make_shared<image_proc::RectifyNode>(options);
+
+    spin_rectify_thread = std::thread([this](){
+      rclcpp::spin(node_rectify);
+    });
 
     // Create raw camera subscriber and publisher
-    image_transport::ImageTransport it(nh_);
+    image_transport::ImageTransport it(node);
     cam_pub_ = it.advertiseCamera(topic_raw_, 1);
   }
 
-  ros::NodeHandle nh_;
+  rclcpp::Node::SharedPtr node;
+  rclcpp::Node::SharedPtr node_rectify;
+  std::thread spin_rectify_thread;
   std::string topic_raw_;
   std::string topic_mono_;
   std::string topic_rect_;
   std::string topic_color_;
   std::string topic_rect_color_;
 
   cv::Mat distorted_image_;
-  sensor_msgs::ImagePtr raw_image_;
+  sensor_msgs::msg::Image::SharedPtr raw_image_;
   bool has_new_image_;
   cv::Mat received_image_;
-  sensor_msgs::CameraInfo cam_info_;
+  sensor_msgs::msg::CameraInfo cam_info_;
   image_transport::CameraPublisher cam_pub_;
   image_transport::Subscriber cam_sub_;
 
 public:
-  void imageCallback(const sensor_msgs::ImageConstPtr & msg)
+  void imageCallback(const sensor_msgs::msg::Image::ConstSharedPtr & msg)
   {
     cv_bridge::CvImageConstPtr cv_ptr;
 
     try {
       cv_ptr = cv_bridge::toCvShare(msg, sensor_msgs::image_encodings::BGR8);
     } catch (cv_bridge::Exception & e) {
-      ROS_FATAL("cv_bridge exception: %s", e.what());
+      RCLCPP_INFO(node->get_logger(), "cv_bridge exception: '%s'", e.what());
```

Comment:
this should at least be RCLCPP_ERROR (I'm not sure there is an RCLCPP_FATAL)

Reply:
https://github.com/ros-perception/image_pipeline/pull/910/commits/f74aa0d71e5016de8d039bf10b00ed36e14c1bd6